### PR TITLE
feat(provider): add WorkOS provider

### PR DESCRIPTION
**Pull request by i-palindrome-i**
## Reasoning 💡

Add WorkOS provider.

### What is WorkOS?
WorkOS is a set of building blocks for quickly adding enterprise features to your app (including SSO), for more information: [https://workos.com/](https://workos.com/)

## Checklist 🧢
- [x] Documentation
- [ ] Tests N/A (manually)
- [x] Ready to be merged

### CODE REVIEW

File: www/docs/providers/workos.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        token.firstName = profile.first_name
+      }
+      return token
+    },
+    async session(session, user) {
+      // add any WorkOS profile properties you want to access client side here
+      session.user.firstName = user.firstName
+      return session
+    }
+  },
```
Comment (balazsorban44) [inline www/docs/providers/workos.md:34]:
```suggestion
```

This is misplaced, should not be inside the providers array, but in any way, I would like the provider examples be specific for the provider. We document the callbacks here already: https://next-auth.js.org/configuration/callbacks

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Nice PR, but had a small comment. Otherwise looks good!

File: www/docs/providers/workos.md (hunk 1 above)
Comment (i-palindrome-i) [reply www/docs/providers/workos.md]: Makes sense, updated.

File: www/docs/providers/workos.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+id: workos
+title: WorkOS
+---
+
+## Documentation
+
+https://workos.com/docs/sso/guide
+
```
Comment (lluia) [inline www/docs/providers/workos.md:9]: Following #1955, can you add this section here, so we don't need to come back and update this file once your PR is merged? 😄 
```suggestion
## Options

The **42 School Provider** comes with a set of default options:

- [42 School Provider options](https://github.com/nextauthjs/next-auth/blob/main/src/providers/42.js)

You can override any of the options to suit your own use case.
```

**Review by 0ubbe — CHANGES_REQUESTED** [review]
Nice ❤️ , one small suggestion 👍🏽

File: www/docs/providers/workos.md (hunk 2 above)
Comment (i-palindrome-i) [reply www/docs/providers/workos.md:9]: Done!

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**